### DISCUSSION TITLE
GLSP vscode client: How to diasable layout editing

### DISCUSSION THREAD

**Discussion by sathya-1994** [eclipse-glsp/glsp#397] [General] [answered]
I am using GLSP's vscode client(https://github.com/eclipse-glsp/glsp-vscode-integration),

Currently we can drag and drop edges and nodes. But I want to disable this layout editing.

**Screenshot:**
![image](https://user-images.githubusercontent.com/59145404/133562177-8f84601d-3981-457c-a10a-d1eb6ad51e24.png)

**Comment by tortmayr** [comment, marked as answer]
Hi, 
you can customize this behavior using the Typehints API. This allows you to define on the server side whether elements of a certain type should be `deleteable`,`repositonable`, `resizeable` etc.
For you usecase you should deactive the `repositonable` flag for all element types.
The type hints are configured in the `DiagramConfiguration` class : https://github.com/eclipse-glsp/glsp-server/blob/c1f7b8d0af052a5ac3ba2adc4590355519509255/examples/org.eclipse.glsp.example.workflow/src/org/eclipse/glsp/example/workflow/WorkflowDiagramConfiguration.java#L70
